volatility3/cli/volshell/generic.py

```diff
@@ -797,6 +798,51 @@ def create_configurable(
 
         return constructed
 
+    def breakpoint(
+        self, address: int, layer_name: Optional[str] = None, lowest: bool = False
+    ) -> None:
+        """Sets a breakpoint on a particular address (within a specific layer)"""
+        if layer_name is None:
+            if self.current_layer is None:
+                raise ValueError("Current layer must be set")
+            layer_name = self.current_layer
+
+        layer: interfaces.layers.DataLayerInterface = self.context.layers[layer_name]
+
+        if lowest:
+            while isinstance(layer, interfaces.layers.TranslationLayerInterface):
+                mapping = layer.mapping(address, 1)
+                if not mapping:
+                    raise ValueError(
+                        "Offset cannot be mapped lower, cannot break at lowest layer"
+                    )
+                _, _, mapped_offset, _, mapped_layer_name = next(mapping)
+                layer = self.context.layers[mapped_layer_name]
+                address = mapped_offset
+
+        # Check if the read value is already overloaded
+        if not hasattr(layer.read, "breakpoints"):
+            # Layer read is not yet wrapped
+            def wrapped_read(offset: int, length: int, pad: bool = False) -> bytes:
+                original_read = getattr(wrapped_read, "original_read")
+                for breakpoint in getattr(wrapped_read, "breakpoints"):
+                    if (offset <= breakpoint) and (breakpoint < offset + length):
+                        import pdb
+
+                        pdb.set_trace()
+                        print("Hit breakpoint")
```

Comment:
I think it's worth flipping the order of the print and the set_trace(), that way the print happens normally, then you're in the python debug.

I think it might be worth changing the print to just to save some future headaches? It makes the print quite long though? 
```
print("Hit breakpoint, entering python debugger. To continue running without the debugger use the command continue")
```